pkg/operator/ceph/cluster/cluster.go

```diff
@@ -478,3 +477,37 @@ func (c *cluster) postMonStartupActions() error {
 
 	return nil
 }
+
+func (c *cluster) configureMsgr2() error {
+	// Set network encryption
+	monStore := config.GetMonStore(c.context, c.ClusterInfo)
+	encryptionSetting := "crc secure"
+	logger.Infof("setting msgr2 encryption mode to %q", encryptionSetting)
+	if c.Spec.Network.ConnectionsEncrypted {
+		encryptionSetting = "secure"
+	}
+	if err := monStore.Set("global", "ms_cluster_mode", encryptionSetting); err != nil {
+		return errors.Wrapf(err, "failed to set ms_cluster_mode to %q", encryptionSetting)
+	}
+	if err := monStore.Set("global", "ms_service_mode", encryptionSetting); err != nil {
+		return errors.Wrapf(err, "failed to set ms_service_mode to %q", encryptionSetting)
+	}
+	if err := monStore.Set("global", "ms_client_mode", encryptionSetting); err != nil {
+		return errors.Wrapf(err, "failed to set ms_client_mode to %q", encryptionSetting)
+	}
```

Comment:
I think it's a bit too "aggressive" to enable all, especially for older client support. I think the spec is a bit too general.

Reply:
Since Rook's goal is to simplify cluster management, I really want to expose only what is needed for the majority of users. Most users either care if everything is encrypted, or are ok that nothing is encrypted on the network. Most users shouldn't worry about whether it's internal ceph traffic, or client traffic. All or nothing should cover most scenarios. If Rook exposes exactly one Ceph setting per setting in the CR, it is usually a sign of too much complexity.

If we expose all the settings, it is much more difficult for users to understand whether they should enable individual settings. If an advanced user really wants to have a different combination of settings, they could just not set this in the CR and then go to the toolbox to configure it with the granular settings.

Reply:
Also made the settings nullable so if it's not set in the cluster CR, they can configure it through the toolbox if needed for more granularily.

Reply:
I agree with Rook's goal for simplicity, and yes we should strive for it. However, the reality is a bit different and I'd argue that Rook should only enable service and cluster, and the client should be left as a CSI setting. So yes, this will increase the complicity a little...

Also even if we don't want to go really granular, I still maintain my request of change for something like:

```
network:
  connections:
    encryption:
      enabled: true
    compression:
      enabled: true
```

With this, we can more nicely add up on the `connections` spec and all the related items will fall under it. 
Also, in the past, we have seen that just having a bool in the spec was not enough as more features were added.

Reply:
> I agree with Rook's goal for simplicity, and yes we should strive for it. However, the reality is a bit different and I'd argue that Rook should only enable service and cluster, and the client should be left as a CSI setting. So yes, this will increase the complicity a little...

If it's a CSI setting, it would be specified in the storage class and this would imply that different storage classes could have different settings for the client encryption, right? I don't see how the Ceph setting can be applied to certain clients and not others, it looks like a cluster-wide setting. Or can you clarify how it would be a CSI setting?

> Also even if we don't want to go really granular, I still maintain my request of change for something like:
> 
> ```
> network:
>   connections:
>     encryption:
>       enabled: true
>     compression:
>       enabled: true
> ```
> 
> With this, we can more nicely add up on the `connections` spec and all the related items will fall under it. Also, in the past, we have seen that just having a bool in the spec was not enough as more features were added.

Agreed on this point, I overlooked it in the response yesterday, thanks.

Reply:
> > I agree with Rook's goal for simplicity, and yes we should strive for it. However, the reality is a bit different and I'd argue that Rook should only enable service and cluster, and the client should be left as a CSI setting. So yes, this will increase the complicity a little...
> 
> If it's a CSI setting, it would be specified in the storage class and this would imply that different storage classes could have different settings for the client encryption, right? I don't see how the Ceph setting can be applied to certain clients and not others, it looks like a cluster-wide setting. Or can you clarify how it would be a CSI setting?

What I meant was a setting for the entire CSI configuration, not per client. I might be thinking too much but a setting in the Operator config for CSI secure client connection?
I guess now that I think of it, I wonder what would happen if the cluster is consumed by external older csi clients and that the client secure connection is enabled in the cluster.
My thought process on this is the following, because Rook deploys and maintains only the "cluster", so it provides the storage resources, it should just configure the service/cluster aspect of it and let the CSI decision elsewhere.
However, there is obviously a buy-in that Rook is tightly coupled with ceph-csi and thus it could make sense to configure it as well. But again I might be thinking too hard here 🤠 , let's discuss a little bit today.

> 
> > Also even if we don't want to go really granular, I still maintain my request of change for something like:
> > ```
> > network:
> >   connections:
> >     encryption:
> >       enabled: true
> >     compression:
> >       enabled: true
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > With this, we can more nicely add up on the `connections` spec and all the related items will fall under it. Also, in the past, we have seen that just having a bool in the spec was not enough as more features were added.
> 
> Agreed on this point, I overlooked it in the response yesterday, thanks.

No worries :)

Reply:
> What I meant was a setting for the entire CSI configuration, not per client. I might be thinking too much but a setting in the Operator config for CSI secure client connection? 

The ceph client encryption setting is a per-ceph cluster setting. So if the operator is managing multiple clusters, they each could have a different encryption setting if it's in the cluster CR. If it's in the operator, we lose that flexibility.

> I guess now that I think of it, I wonder what would happen if the cluster is consumed by external older csi clients and that the client secure connection is enabled in the cluster. My thought process on this is the following, because Rook deploys and maintains only the "cluster", so it provides the storage resources, it should just configure the service/cluster aspect of it and let the CSI decision elsewhere. However, there is obviously a buy-in that Rook is tightly coupled with ceph-csi and thus it could make sense to configure it as well. But again I might be thinking too hard here 🤠 , let's discuss a little bit today.

Yes, let's discuss. My thought is that the admin of the cluster owns whether all connections to the cluster must be secure. If there is an older client, they wouldn't be able to connect. Time to upgrade if they want to connect! :)